### ISSUE TITLE
FileList is iterable

### ISSUE THREAD

**Issue by jimmywarting** [google/closure-compiler#3231]
I just tried [this](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F**%250A%2520*%2520%2540param%2520%257BFileList%257D%2520list%250A%2520*%252F%250Afunction%2520hello(list)%2520%257B%250A%2520%2520console.log(...list)%250A%257D%250A):

```js
/**
 * @param {FileList} list
 */
function hello(list) {
  console.log(...list)
}
```
Followed by this warning

```
Spread operator only applies to Iterable types
```

Well, it's iterable... I don't know if this is related to: [idlharness: add iterable members if implicitly iterable](https://github.com/web-platform-tests/wpt/issues/8616)? those it mean that everything that is arraylike should also be iterable?

**Comment by tjgq** [comment]
My reading of the spec is that [`FileList`](https://w3c.github.io/FileAPI/#filelist-section) isn't iterable (compare e.g. with [`NodeList`](https://w3c.github.io/dom/#nodelist), which is), even though some browsers implement it.
